docs/AddingNewProjects.md

```diff
@@ -0,0 +1,48 @@
+# Adding New Projects To The Repo
+
+Sample PR of final result: https://github.com/dotnet/aspnetcore/pull/41945
+
+## Creating a new project
+1. Create a new folder that will house your `.csproj` and other project-related files.
+2. (EXTREMELY IMPORTANT) Inside our new folder make a new folder called `src`.
+3. In VS, Add a new `Solution Folder` in the new folder from the first step.
+4. Create the project via the VS `Add` menu (select the folder -> right click -> Add -> follow the wizard).
+
+    **Note:** Depending on what kind of project you are creating, VS will create different files in your project. You might also want to add the following files:
+    - `PublicAPI.Shipped.txt`
+      - Lists publically visible APIs that are exported from your final compiled `.dll`.
+      - This only lists APIs that have already been shipped to customers and cannot be changed.
+      - There is an empty template at `eng/PublicAPI.empty.txt` for your reference.
+    - `PublicAPI.UnShipped.txt`
+      - Lists publicly visible APIs that are exported from your final compiled `.dll`. If this is not configured properly, you will get build errors. VS will warn you though with green squiggly lines. If you see these squiggly lines, open the VS Quick Actions (CTRL + '.') and select the option to and it to the public API.
+      - This only lists APIs that have NOT already been shipped to customers. So, these can still change.
+      - There is an empty template at `eng/PublicAPI.empty.txt` for your reference.
+    - `AssemblyInfo.cs`
+      - Lists various properties of your compiled assembly, such as to which other packages your `internal` properties and methods are available to.
+      - You can also expose internals via the `@(InternalsVisibleTo)` item in your project file instead of using this `AssemblyInfo.cs` file.
```

Comment:
nit
```suggestion
      - You can also expose internals via `@(InternalsVisibleTo)` items in your project file instead of using this `AssemblyInfo.cs` file.
```